src/emulator.py

```diff
@@ -252,34 +253,79 @@ def read_and_confirm_mnemonic() -> None:
         client.input(mnemonic[index])
         time.sleep(SLEEP)
 
+    # Click Continue to finish the quiz
     client.press_yes()
+    time.sleep(SLEEP)
+
+    # Click Continue to finish the backup
     client.press_yes()
+    time.sleep(SLEEP)
+
     client.close()
 
 
-def read_and_confirm_mnemonic_shamir():
+def read_and_confirm_mnemonic_shamir(shares: int = 1, threshold: int = 1):
+    # TODO: decide what numbers will be allowed (there are some maximum numbers)
+    if shares < 1 or shares > 6:
+        raise RuntimeError("Number of shares must be between 1 and 6.")
+    if threshold > shares:
+        raise RuntimeError("Threshold cannot be bigger than number of shares.")
+
+    # For setting the right amount of shares/thresholds, we need location of buttons
+    MINUS_BUTTON_COORDS = (60, 70)
+    PLUS_BUTTON_COORDS = (180, 70)
+
+    # Connecting to the device
     client = DebugLink(get_device().find_debug())
     client.open()
     time.sleep(SLEEP)
 
     # Click Continue to begin Shamir setup process
     client.press_yes()
+    time.sleep(SLEEP)
 
-    # Clicking the minus button four times to set only one share (it starts at 5)
-    minus_button_coords = (60, 70)
-    for _ in range(4):
-        client.click(minus_button_coords)
-        time.sleep(SLEEP)
-
-    # Click Continue to confirm one share
+    # Clicking the minus/plus button to set right number of shares (it starts at 5)
+    default_shares = 5
+    needed_clicks = abs(shares - default_shares)
+    if needed_clicks > 0:
+        if shares < default_shares:
+            button_coords_to_click = MINUS_BUTTON_COORDS
+        else:
+            button_coords_to_click = PLUS_BUTTON_COORDS
+        for _ in range(needed_clicks):
+            client.click(button_coords_to_click)
+            time.sleep(SLEEP)
+
+    # Click Continue to confirm the number of shares
     client.press_yes()
     time.sleep(SLEEP)
 
     # Click Continue to set threshold
     client.press_yes()
     time.sleep(SLEEP)
 
-    # Click Continue to set threshold to one
+    # When we have 1 or 2 shares, the threshold is set and cannot be changed
+    # (threshold will be 1 and 2 respectively)
+    # Otherise assign it correctly
+    if shares not in [1, 2]:
+        # TODO: extend this for all supported numbers in the future
+        default_thresholds_for_shares_on_screen = {
+            3: 2,
+            4: 3,
+            5: 3,
+            6: 4,
+        }
```

Comment:
We can't read this somehow via the debuglink as well right?

Reply:
Good idea. Also, it seems to follow a pattern - `(x//2) + 1`, but we should verify it. I will try to take it from device, if there

Reply:
I could not find it written in the device directly, but [firmware repo](https://github.com/trezor/trezor-firmware/blob/master/core/src/apps/management/reset_device/layout.py#L272) confirms the equation above, so I will use that

Reply:
Great :+1: :+1: